### ISSUE TITLE
Intelisense unusably slow for JS/TS 

### ISSUE THREAD

**Issue by henrique-smr** [microsoft/vscode#118393]
Issue Type: <b>Performance Issue</b>

I sometimes have to wait up to minutes before any response from hovers/completions.

Any feature related to the Language Service, as ESlint for example, takes ages to react to any changes.

VS Code version: Code 1.54.1 (f30a9b73e8ffc278e71575118b6bf568f04587c8, 2021-03-04T22:38:50.094Z)
OS version: Linux x64 5.4.0-66-generic snap

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-8265U CPU @ 1.60GHz (8 x 3699)|
|GPU Status|2d_canvas: unavailable_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: disabled_off<br>protected_video_decode: disabled_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|2, 2, 1|
|Memory (System)|7.66GB (0.23GB free)|
|Process Argv|--no-sandbox --crash-reporter-id 2edd6361-92bf-4018-8ff9-e56533b54582|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|xfce|
|XDG_CURRENT_DESKTOP|XFCE|
|XDG_SESSION_DESKTOP|xfce|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    0	   133	  6079	code main
    0	    39	  6081	   zygote
    0	    71	  6128	     gpu-process
    0	    39	  6082	   zygote
    0	    78	  6807	     window (Issue Reporter)
    0	    63	  6112	   utility
    0	   337	  6137	   window (Identidade.js - FatApp (Workspace) - Visual Studio Code)
    0	   102	  6161	     extensionHost
    0	   118	  6218	       electron_node tsserver.js 
   12	   376	  6219	       electron_node tsserver.js 
    0	    86	  6360	         electron_node typingsInstaller.js typesMap.js 
    0	    63	  6458	       /snap/code/57/usr/share/code/code /snap/code/57/usr/share/code/resources/app/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=6161
    0	   110	  6186	     watcherService
    1	   118	  6173	   shared-process
    0	    63	  6248	     ptyHost
    0	     0	  6371	       /bin/bash
    0	     0	  6825	     /bin/sh -c /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	     0	  6826	       /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (Identidade.js - FatApp (Workspace) - Visual Studio Code)
|    Folder (FatApp): 11483 files
|      File types: py(1979) js(1954) pyc(1763) html(1183) jpg(536) scss(287)
|                  png(236) json(168) ttf(167) md(147)
|      Conf files: package.json(29) gulp.js(11) jsconfig.json(4) tasks.json(2)
|                  tsconfig.json(2) launch.json(1) grunt.js(1)
|      Launch Configs: node-terminal pwa-chrome;
```

</details>
<details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
material-theme|zhu|3.9.15

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt875:30259475
dockersubset:30265998
pythontb:30265425
vspre833cf:30267465

```

</details>

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/) with all [extensions disabled](https://github.com/microsoft/vscode/wiki/Performance-Issues#run-with-extensions-disabled)?

**Comment by henrique-smr** [comment]
Yes. It does.

The funny thing is that for some files it wont happen. But as soon as I open one that does break Intelisense, I have to restart Vscode to get it functioning again.

I try to get some information within the Logs of tsserver (verbose mode on), but couldn't get much out of it. It appears that tsserve just become irresponsive.

I'm not a Language Server guy, and I would appreciate really much if someone could point out what could be causing this, as is not great for productivity

**Comment by mjbvz** [comment]
Does this only happen for a specific file/project? If so, can you please share that project (or, even better, a stripped version of it that demonstrates the issue)

**Comment by henrique-smr** [comment]
This is a private project from my company, but i can see with my colleagues what we can share.

This happens to some files only, and for those that are more deep in the dependency tree.

I don't know if it helps, but we have a way to share commonly used components/providers/utils over the different parts of the project, that is to create symbolic links in a node_modules folder which is inside of the 'src' folder, besides the normal node_modules, which contains all the third party modules and sits on the root of.

That way we avoid giant relative parent imports ('../../../../../example')

Perhaps this is messing with the tsserver?

**Comment by henrique-smr** [comment]
This project has ~250 files and ~500 inside dependencies (not a third party dependency).

**Comment by henrique-smr** [comment]
Those are our outside dependencies:

 "dependencies": {
    "axios": "^0.19.2",
    "firebase": "^7.21.1",
    "moment": "^2.29.0",
    "prop-types": "^15.7.2",
    "react": "^16.13.1",
    "react-datepicker": "^3.2.2",
    "react-dom": "^16.13.1",
    "react-feather": "^2.0.8",
    "react-input-mask": "^2.0.4",
    "react-number-format": "^4.4.1",
    "react-router-dom": "^5.2.0",
    "react-scripts": "3.4.3",
    "react-select": "^3.1.0",
    "react-window": "^1.8.5"
  },

**Comment by henrique-smr** [comment]
We could share some code if it was done privately. Is any way we could achieve this?

Thanks

**Comment by mjbvz** [comment]
Please try putting together a minimal project. If you still aren't comfortable sharing that publicly, you can email it to me: matb at Microsoft

**Comment by dcastil** [comment]
I have a similar issue since `v1.54.1`. ESLint works fine but the TS intellisense gets very slow sometimes. I couldn't find a way to reproduce it properly just yet, but I'll post here if I find out more about it.

So far my observations are:
* It depends on the file. There are files in which intellisense is fast, but others are very slow. I only tested `.tsx` files.
* It doesn't depend on the project. I had both fast and slow files in a single project.
* Restarting VS Code didn't change anything
* These issues weren't present in previous versions of VS Code in the same files

I made two videos which illustrate the issue:

Slow intellisense and syntax highlighting

https://user-images.githubusercontent.com/31006608/110663598-e0cd7100-81c6-11eb-8ed3-c68f74464928.mov

Very slow TS server

https://user-images.githubusercontent.com/31006608/110663646-ec209c80-81c6-11eb-88c4-61a8d7dfbe79.mov
